apps/files/src/FileUpload.js

```diff
@@ -12,7 +12,11 @@ class FileUpload {
     const xhr = new XMLHttpRequest()
 
     // Headers
-    xhr.open(this.type, this.url + encodeURIComponent(this.path), true)
+    this.path = this.path.replace(/(^\/+)/, '')
```

Comment:
Better do this in the ctor? On a second call to upload the path will be encoded again

Reply:
the fileupload object is getting created for every upload call anyway.
Path can have multiple values. e.g.: Images/Landscape/1.png Images/Sea/1.png - means it has to be done anyway somewhere